src/main/java/org/mapfish/print/output/AbstractOutputFormat.java

```diff
@@ -31,7 +34,19 @@ protected RenderingContext doPrint(PrintParams params) throws DocumentException
 
         doc.close();
         writer.close();
-
+        
+        // close all opened PdfReader's
+        List<PdfReader> readerList = context.getCustomBlocks().getReaders();
+        if (readerList != null) {
```

Comment:
Cannot be null, this if is useless.

Reply:
And you could move that code to `PDFCustomBlocks`...

Reply:
- null check is not useless, because you will only add PdfReader to the list when you have a backgroundPdf. If we don't have a backgroundPdf, we don't have this problem of open file, so we don't need to close the templates.
- yes, we can move the code to PDFCustomBlocks to a method, and then call that method at the AbstractOutputFormat after closing the document and writer.

Reply:
No, you always initialise the readers attribute in the constructor. So the attribute is never null. I think you mixup a null list with an empty list. Not the same thing.